File: EclipseKotlin/Kotlin/src/main/kotlin/fun/fizzy/patterns/PatternsMain.kt
package `fun`.fizzy.patterns

private val mPatternsDriver: PatternsDriver = PatternsDriver()

fun main() {
	println("* * * * * Observer pattern. * * * * *")
	val oo = ObservingObject()
	oo.addObserver(mPatternsDriver)
	oo.notifyObservers("I am a message passed through the observer and will succeed.")
	oo.removeObserver(mPatternsDriver)
	oo.notifyObservers("I am a message passed through the observer and will fail.")
}

File: EclipseCpp/Cpp/src/algorithms/patterns/PatternsMain.cpp

#include "Patterns.h"

int main() {
    cout << "* * * * * Observer pattern. * * * * *" << endl;
    ObservingObject observingObject;
    auto patternsDriver = std::make_shared<PatternsDriver>();
    observingObject.addObserver(patternsDriver);
    observingObject.notifyObservers("I am a message passed through the observer and will succeed.");
    observingObject.removeObserver(patternsDriver);

    return 0;
}